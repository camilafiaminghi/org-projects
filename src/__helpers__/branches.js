export default [
	{
		"name": "0.3-stable",
		"commit": {
			"sha": "f7c56d77dfbb1f7b8217f04e00ff71a26537c9a8",
			"url": "https://api.github.com/repos/facebook/react/commits/f7c56d77dfbb1f7b8217f04e00ff71a26537c9a8"
		},
		"protected": true
	},
	{
		"name": "0.4-stable",
		"commit": {
			"sha": "283f4a8b04db6bbd177e94404476cabaf569d3a7",
			"url": "https://api.github.com/repos/facebook/react/commits/283f4a8b04db6bbd177e94404476cabaf569d3a7"
		},
		"protected": true
	},
	{
		"name": "0.5-stable",
		"commit": {
			"sha": "7b809ffbc4cdbb9858ba36ec341b4767c7a8543e",
			"url": "https://api.github.com/repos/facebook/react/commits/7b809ffbc4cdbb9858ba36ec341b4767c7a8543e"
		},
		"protected": true
	},
	{
		"name": "0.8-stable",
		"commit": {
			"sha": "56517a54234379239113a663d2805f75a38b11ab",
			"url": "https://api.github.com/repos/facebook/react/commits/56517a54234379239113a663d2805f75a38b11ab"
		},
		"protected": true
	},
	{
		"name": "0.9-stable",
		"commit": {
			"sha": "b1f66edf4e90527468987eafbaf50a410febdcc9",
			"url": "https://api.github.com/repos/facebook/react/commits/b1f66edf4e90527468987eafbaf50a410febdcc9"
		},
		"protected": true
	},
	{
		"name": "0.10-stable",
		"commit": {
			"sha": "d5b409002b425d2242da0d9bacfad68278856400",
			"url": "https://api.github.com/repos/facebook/react/commits/d5b409002b425d2242da0d9bacfad68278856400"
		},
		"protected": true
	},
	{
		"name": "0.11-stable",
		"commit": {
			"sha": "f6fcf385c9fd0df2ad666bc7765c127bd5e1528d",
			"url": "https://api.github.com/repos/facebook/react/commits/f6fcf385c9fd0df2ad666bc7765c127bd5e1528d"
		},
		"protected": true
	},
	{
		"name": "0.12-stable",
		"commit": {
			"sha": "0fc0871c89963ecaeb1476b82156c2399e765e8e",
			"url": "https://api.github.com/repos/facebook/react/commits/0fc0871c89963ecaeb1476b82156c2399e765e8e"
		},
		"protected": true
	},
	{
		"name": "0.13-stable",
		"commit": {
			"sha": "76c87da026bdab63b5b109e3c073a1db74896ed6",
			"url": "https://api.github.com/repos/facebook/react/commits/76c87da026bdab63b5b109e3c073a1db74896ed6"
		},
		"protected": true
	},
	{
		"name": "0.14-stable",
		"commit": {
			"sha": "894d20744cba99383ffd847dbd5b6e0800355a5c",
			"url": "https://api.github.com/repos/facebook/react/commits/894d20744cba99383ffd847dbd5b6e0800355a5c"
		},
		"protected": true
	},
	{
		"name": "15-stable",
		"commit": {
			"sha": "571a9208d5133e8737f565fe60b762d201f0d37c",
			"url": "https://api.github.com/repos/facebook/react/commits/571a9208d5133e8737f565fe60b762d201f0d37c"
		},
		"protected": true
	},
	{
		"name": "15.6-dev",
		"commit": {
			"sha": "5530c7565b729894276b959a0aa44e1487080a65",
			"url": "https://api.github.com/repos/facebook/react/commits/5530c7565b729894276b959a0aa44e1487080a65"
		},
		"protected": false
	},
	{
		"name": "16.3-dev",
		"commit": {
			"sha": "0c756fb33f091cb87afa38fade048ad15a4a2466",
			"url": "https://api.github.com/repos/facebook/react/commits/0c756fb33f091cb87afa38fade048ad15a4a2466"
		},
		"protected": false
	},
	{
		"name": "16.4.2-dev",
		"commit": {
			"sha": "54adb2674afe16ec603e0c54bf46ccf1afa42308",
			"url": "https://api.github.com/repos/facebook/react/commits/54adb2674afe16ec603e0c54bf46ccf1afa42308"
		},
		"protected": false
	},
	{
		"name": "facts",
		"commit": {
			"sha": "babe6d094059ac0837d216913f4e6e0794b14b99",
			"url": "https://api.github.com/repos/facebook/react/commits/babe6d094059ac0837d216913f4e6e0794b14b99"
		},
		"protected": false
	},
	{
		"name": "gaearon-patch-1",
		"commit": {
			"sha": "aa5b748b0313fdda1cc3b58d56a75348d813f2d9",
			"url": "https://api.github.com/repos/facebook/react/commits/aa5b748b0313fdda1cc3b58d56a75348d813f2d9"
		},
		"protected": false
	},
	{
		"name": "gaearon-patch-2",
		"commit": {
			"sha": "71e73f4e65236761405c00e40aaef5c40c7e6939",
			"url": "https://api.github.com/repos/facebook/react/commits/71e73f4e65236761405c00e40aaef5c40c7e6939"
		},
		"protected": false
	},
	{
		"name": "gh-pages",
		"commit": {
			"sha": "d027eca037f06f682f667597bd68a0cd638e4f2e",
			"url": "https://api.github.com/repos/facebook/react/commits/d027eca037f06f682f667597bd68a0cd638e4f2e"
		},
		"protected": true
	},
	{
		"name": "is-renderable",
		"commit": {
			"sha": "efe8e670af1ee930447d169795cabbcbeabb8b1a",
			"url": "https://api.github.com/repos/facebook/react/commits/efe8e670af1ee930447d169795cabbcbeabb8b1a"
		},
		"protected": false
	},
	{
		"name": "lazy-raf-warning",
		"commit": {
			"sha": "a756039cb04adf85bd5d7966f686334ac7889c02",
			"url": "https://api.github.com/repos/facebook/react/commits/a756039cb04adf85bd5d7966f686334ac7889c02"
		},
		"protected": false
	},
	{
		"name": "lazy-wip",
		"commit": {
			"sha": "0535278dea8b535f57f4bab02ad977bb2b878988",
			"url": "https://api.github.com/repos/facebook/react/commits/0535278dea8b535f57f4bab02ad977bb2b878988"
		},
		"protected": false
	},
	{
		"name": "master",
		"commit": {
			"sha": "c11015ff4f610ac2924d1fc6d569a17657a404fd",
			"url": "https://api.github.com/repos/facebook/react/commits/c11015ff4f610ac2924d1fc6d569a17657a404fd"
		},
		"protected": true
	},
	{
		"name": "poc-event-registration",
		"commit": {
			"sha": "2f5d455e44343aaf4a6a079e34b0e2bf569088b5",
			"url": "https://api.github.com/repos/facebook/react/commits/2f5d455e44343aaf4a6a079e34b0e2bf569088b5"
		},
		"protected": false
	},
	{
		"name": "radio-click",
		"commit": {
			"sha": "f0664d552ccc5a6f03864f664bea6a646cf1e521",
			"url": "https://api.github.com/repos/facebook/react/commits/f0664d552ccc5a6f03864f664bea6a646cf1e521"
		},
		"protected": false
	},
	{
		"name": "revert-13535-simpify-some-logic",
		"commit": {
			"sha": "2d1f7c87a068f597d894cd55389fa542a0118fdd",
			"url": "https://api.github.com/repos/facebook/react/commits/2d1f7c87a068f597d894cd55389fa542a0118fdd"
		},
		"protected": false
	},
	{
		"name": "revert-14218-memory-leak-on-unmount",
		"commit": {
			"sha": "9a91f12002f2583ccadc83d5889a91d11334d7b4",
			"url": "https://api.github.com/repos/facebook/react/commits/9a91f12002f2583ccadc83d5889a91d11334d7b4"
		},
		"protected": false
	},
	{
		"name": "revert-14643-fun-sm-hooks",
		"commit": {
			"sha": "e4a8ff98c6a2bafdc96e04ede49f60cc31ee1f11",
			"url": "https://api.github.com/repos/facebook/react/commits/e4a8ff98c6a2bafdc96e04ede49f60cc31ee1f11"
		},
		"protected": false
	},
	{
		"name": "revert-14648-ctx-bad",
		"commit": {
			"sha": "820445c3ce712ad44695f7eec43667c3affa4340",
			"url": "https://api.github.com/repos/facebook/react/commits/820445c3ce712ad44695f7eec43667c3affa4340"
		},
		"protected": false
	},
	{
		"name": "yeswork",
		"commit": {
			"sha": "4c67be212d44a1cdae2f595d43659ee64e4f1843",
			"url": "https://api.github.com/repos/facebook/react/commits/4c67be212d44a1cdae2f595d43659ee64e4f1843"
		},
		"protected": false
	}
]
