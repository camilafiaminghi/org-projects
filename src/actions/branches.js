export const BRANCHES_REQUEST = 'BRANCHES_REQUEST'
export const BRANCHES_SUCCESS = 'BRANCHES_SUCCESS'
export const BRANCHES_FAIL = 'BRANCHES_FAIL'

export function getBranches (org, repo) {
	return {
		type: BRANCHES_REQUEST,
		payload: {
			org,
			repo
		}
	}
}

export function getBranchesSuccess (branches) {
	return {
		type: BRANCHES_SUCCESS,
		payload: branches
	}
}

export function getBranchesFail (error) {
	return {
		type: BRANCHES_FAIL,
		error
	}
}
