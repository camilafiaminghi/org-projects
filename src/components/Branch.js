import React from 'react'
import PropTypes from 'prop-types'

const Branch = ({ name, commit }) => {
	return (
		<div>
			<p>{name}</p>
			<a
				href={commit.url}
				target="_blank">{commit.sha}</a>
		</div>
	)
}

Branch.propTypes = {
  name: PropTypes.string,
  commit: PropTypes.object
}

export default Branch
