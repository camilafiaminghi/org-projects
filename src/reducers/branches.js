import { BRANCHES_REQUEST, BRANCHES_SUCCESS, BRANCHES_FAIL } from '../actions/branches'

const branches = (state = {}, action) => {
	switch (action.type) {
		case BRANCHES_REQUEST:
			return {
				...state,
				org: action.payload,
				loading: true
			}
		case BRANCHES_SUCCESS:
			return {
				...state,
				items: action.payload,
				errors: null,
				loading: false
			}
		case BRANCHES_FAIL:
			return {
				...state,
				errors: action.errors,
				loading: false
			}
		default:
			return state
	}
}

export default branches
