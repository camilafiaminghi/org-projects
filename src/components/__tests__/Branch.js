import React from 'react'
import renderer from 'react-test-renderer'
import Branch from './../Branch'
import branches from './../../__helpers__/branches'

const branch = branches[0]
let router
let wrapper

describe('<Branch />', () => {
	beforeEach(() => {
		wrapper = shallow(<Branch {...branch} />)
	})

	it('matches snapshot', () => {
		const component = renderer.create(<Branch {...branch} />)
		expect(component.toJSON()).toMatchSnapshot()
	})

	it('should render', () => {
		expect(wrapper.exists()).toBeTruthy()
	})

	it('should has a name', () => {
		expect(wrapper.find('p').exists()).toBeTruthy()
		expect(wrapper.find('p').shallow().props().children).toEqual(branch.name)
	})

	it('should has a link to branch url', () => {
		expect(wrapper.find('a').exists()).toBeTruthy()
		expect(wrapper.find('a').props().href).toEqual(branch.commit.url)
		expect(wrapper.find('a').props().children).toEqual(branch.commit.sha)
	})
})
