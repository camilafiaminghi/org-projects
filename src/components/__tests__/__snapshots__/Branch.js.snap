// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Branch /> matches snapshot 1`] = `
<div>
  <p>
    0.3-stable
  </p>
  <a
    href="https://api.github.com/repos/facebook/react/commits/f7c56d77dfbb1f7b8217f04e00ff71a26537c9a8"
    target="_blank"
  >
    f7c56d77dfbb1f7b8217f04e00ff71a26537c9a8
  </a>
</div>
`;
